3.12-24.04/rockcraft.yaml

```
@@ -0,0 +1,85 @@
+name: rabbitmq
+summary: RabbitMQ is an open source multi-protocol messaging broker.
+description: |
+  RabbitMQ is a reliable and mature messaging and streaming broker, which is
+  easy to deploy on cloud environments, on-premises, and on your local machine.
+  It is currently used by millions worldwide. This is a chiselled RabbitMQ
+  image.
+version: "3.12"
+license: Apache-2.0
+base: bare
+build-base: ubuntu@24.04
+platforms:
+  amd64:
+  arm64:
+run-user: _daemon_
+
+parts:
+  rabbitmq:
+    plugin: nil
+    stage-packages:
+      - rabbitmq-server_bins
+      - base-files_chisel
+      - dash_bins
+      - erlang-base_bins
+      - erlang-crypto_modules
+      - erlang-eldap_modules
+      - erlang-inets_scripts
+      - erlang-mnesia_modules
+      - erlang-os-mon_bins
+      - erlang-parsetools_modules
+      - erlang-public-key_modules
+      - erlang-runtime-tools_modules
+      - erlang-ssl_modules
+      - erlang-syntax-tools_modules
+      - erlang-tools_modules
+      - erlang-xmerl_modules
+
+  set-user:
+    plugin: nil
+    after: [rabbitmq]
+    override-stage: |
+      mkdir -p ${CRAFT_STAGE}/var/lib/rabbitmq/mnesia ${CRAFT_STAGE}/var/log/rabbitmq \
+               ${CRAFT_STAGE}/etc/rabbitmq/conf.d
+      chown -R 584792:584792 ${CRAFT_STAGE}/var/lib/rabbitmq ${CRAFT_STAGE}/var/log/rabbitmq\
+                       ${CRAFT_STAGE}/etc/rabbitmq
```

```suggestion
      chown -R 584792:584792 ${CRAFT_STAGE}/var/lib/rabbitmq \
          ${CRAFT_STAGE}/var/log/rabbitmq \
          ${CRAFT_STAGE}/etc/rabbitmq
```
nitpick ^^

Also, is the 584792 UID:GID of user `_daemon_`? If so, and I am not sure if it possible, what do you think about `chown -R _daemon_:_daemon_ ..`? I don't know what the best practice for this is though, in rocks.

---

Yes. It's written in rockcraft's documentation. No, the user and group `_daemon_` are not known to the builder, and they are only available in the final image, so we have to give the numbers here explicitly.

```
'override-build' in part 'configs' failed with code 1.
Detailed information: 
:: + cp -r etc/ /root/parts/configs/install
:: + chown -R _daemon_:_daemon_ /root/parts/configs/install/etc/rabbitmq
:: chown: invalid user: ‘_daemon_:_daemon_’
Recommended resolution: Review the scriptlet and make sure it's correct.
```

---

Ah I see! Thanks for the explanation!